src/v/pandaproxy/schema_registry/sharded_store.h

```diff
@@ -112,27 +118,27 @@ class sharded_store {
     /// reads to the topic.
     template<typename F>
     ss::future<> sync(F f) {
-        auto sem_units = co_await ss::smp::submit_to(
-          ss::shard_id{0},
-          _smp_opts,
-          std::move(
-            [this]()
-              -> ss::future<ss::foreign_ptr<std::unique_ptr<ss::semaphore_units<
-                ss::semaphore_default_exception_factory,
-                ss::timer<>::clock>>>> {
-                co_return ss::make_foreign(
-                  std::make_unique<ss::semaphore_units<
-                    ss::semaphore_default_exception_factory,
-                    ss::timer<>::clock>>(co_await ss::get_units(_sync_sem, 1)));
-            }));
+        auto sem_units = co_await get_units_foreign(
```

Comment:
Aaargh.  Thank you.  I was getting confused about why the compiler wouldn't let me use my sem_units alias in my template arguments, and the mess was all because I hadn't noticed my variable name overlapped with the type.  Tidied this up.